dali/image/jpeg2k.cc

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/image/jpeg2k.h"
+#include "dali/core/span.h"
+
+namespace dali {
+
+using block_type_t = std::array<uint8_t, 4>;
+
+namespace {
+
+constexpr block_type_t jp2_sig_type = {'j', 'P', ' ', ' '};
+
+constexpr block_type_t jp2_format_type = {'f', 't', 'y', 'p'};
+
+constexpr block_type_t jp2_header_type = {'j', 'p', '2', 'h'};
+
+constexpr block_type_t jp2_im_header_type = {'i', 'h', 'd', 'r'};
+
+template <typename T, int B = sizeof(T)>
```

Comment:
I would add check if T is unsigned (https://en.cppreference.com/w/cpp/types/is_signed).

Reply:
done